src/Histogram.js

```diff
@@ -308,44 +368,32 @@ export default class Histogram {
     if (div > 0) {
       var lut = new Uint8Array(256 * 4);
 
-      // compute lut as if continuous
-      for (let i = 0; i < lut.length / 4; ++i) {
+      // compute lut and track control points for the piecewise linear sections
+      const lutControlPoints = [{ x: 0, opacity: 0, color: [255, 255, 255] }];
+      lut[0] = 255;
+      lut[1] = 255;
+      lut[2] = 255;
+      lut[3] = 0;
+      let slope = 0;
+      let lastSlope = 0;
+      let opacity = 0;
+      let lastOpacity = 0;
+      for (let i = 1; i < lut.length / 4; ++i) {
         lut[i * 4 + 0] = 255;
         lut[i * 4 + 1] = 255;
         lut[i * 4 + 2] = 255;
-        lut[i * 4 + 3] = Math.clamp(255 * ((map[i] - map[0]) / div), 0, 255);
-      }
-
-      // compute control points piecewise linear.
-      const lutControlPoints = [{ x: 0, opacity: 0, color: [255, 255, 255] }];
-      // read up to the first nonzero.
-      let i = 1;
-      for (i = 1; i < map.length; ++i) {
-        if (map[i] > 0) {
-          lutControlPoints.push({
-            x: i - 1,
-            opacity: 0,
-            color: [255, 255, 255],
-          });
-          break;
+        lastOpacity = opacity;
+        opacity = Math.clamp(Math.round(255 * (map[i] - map[0])), 0, 255);
+        lut[i * 4 + 3] = opacity;
+
+        slope = opacity - lastOpacity;
+        // if map[i]-map[i-1] is the same as map[i+1]-map[i] then we are in a linear segment and do not need a new control point
+        if (slope != lastSlope) {
+          lutControlPoints.push({ x: i - 1, opacity: lastOpacity / 255.0, color: [255, 255, 255] });
```

Comment:
How come the color is always 255, 255, 255 here? Is that what "equalized" lookup table means?

Reply:
The preset functions that generate lookup tables don't know anything about color.  They all use 255,255,255 for the color (except for the brand new colorize one) and only really modulate opacity.  To be honest, the Equalize function doesn't get used by any client code right now but it was one where the LUT and the control points were incompatible (based on unit tests!)  So I put in this change to fix that.